### PR TITLE
fix replace missing metadata for music

### PR DESCRIPTION
**Pull request by bhowe34**
Continuing the work started in https://github.com/jellyfin/jellyfin/pull/9985 which appears to be stalled or abandoned.

**Changes**
Ensures user provided metadata for music is not overwritten when only searching for missing metadata. Also, check if the name field is locked before updating the value.

**Issues**
Fixes issue https://github.com/jellyfin/jellyfin/issues/9760 for music.

### CODE REVIEW

**Review by JPVenson — APPROVED** [review]

**Review by MarkCiliaVincenti — APPROVED** [review]

**Comment by bhowe34** [comment]
Anything I can do to help move this along? It still needs one more approval before I can merge.

**Comment by JPVenson** [comment]
@bhowe34 not really. Just wait until one other core member has time to look at it. Also you cannot merge it on your own but you have to wait for one of the other core member to merge it then.

**Review by crobibero — APPROVED** [review]

**Pull request merged**
